File: php/symmetric-tree.php
<?php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {

    /**
     * @param TreeNode $root
     * @return Boolean
     */
    function isSymmetric($root) {
        if($root->left == null && $root->right == null) {
            return true;
        }
        return $this->checkForSymmetry($root->left, $root->right, $flag);
    }
    
    function checkForSymmetry($root1, $root2, &$flag) {
        if(($root1 == null && $root2 != null) || ($root1 != null && $root2 == null)) {
            $flag = false;
            return $flag;
        } else if($root1 == null && $root2 == null) {
            return true;
        } else {
            if($root1->val != $root2->val) {
                $flag = false;
                return $flag;
            } else {
                $flag = true;
            }
            if(!($this->checkForSymmetry($root1->left, $root2->right, $flag)))
                return $flag;
            $this->checkForSymmetry($root1->right, $root2->left, $flag);
        }
        return $flag;
    }
}
?>

File: Go/SymmetricTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    if root == nil || (root.Left == nil && root.Right == nil) {
        return true
    }

    return checkForSymmetry(root.Left, root.Right)
}

func checkForSymmetry(leftTree *TreeNode, rightTree *TreeNode) bool {
    if leftTree == nil && rightTree == nil {
        return true
    }
    if leftTree == nil || rightTree == nil {
        return false
    }

    leftCheck := checkForSymmetry(leftTree.Left, rightTree.Right)
    rightCheck := checkForSymmetry(leftTree.Right, rightTree.Left)
    
    return (leftTree.Val == rightTree.Val)&&leftCheck&&rightCheck
}
